testsuite/tests/typing-layouts/basics.ml

```diff
@@ -2857,3 +2857,28 @@ Error: This expression has type "float#" but an expression was expected of type
        But the layout of float# must be a sublayout of value
          because it's the type of the recursive variable x.
 |}]
+
+(**********************************************************)
+(* Test 46: errors when functions don't have layout value *)
+
+let f (x : ('a : bits64)) = x ()
+
+[%%expect{|
+Line 1, characters 28-29:
+1 | let f (x : ('a : bits64)) = x ()
+                                ^
+Error: This expression is used as a function, but its type "'a"
+       has kind "bits64", which cannot be the kind of a function.
+       (Functions always have kind "value mod unique uncontended".)
```

Comment:
Should we print the full `value mod unique uncontended` here?

Reply:
Probably, there's nothing wrong with that, but maybe we can do some nice pretty-printing. I think it's not worth doing, but maybe you disagree.

Reply:
I don't have a shorter way to print that. The error should be possible only with a kind annotation, so the programmer at least knows about kinds. I think it's OK.